exasol/nb_connector/text_ai_extension_wrapper.py

```diff
@@ -1,12 +1,57 @@
-from typing import Optional
+from typing import Optional, Generator
+from contextlib import contextmanager
+from pathlib import Path
+import requests
+import subprocess
+import tempfile
 
 from exasol.nb_connector.secret_store import Secrets
-from pathlib import Path
+from exasol.nb_connector.ai_lab_config import AILabConfig as CKey
 
 LANGUAGE_ALIAS = "PYTHON3_TXAIE"
 
 LATEST_KNOWN_VERSION = "???"
 
+
+@contextmanager
+def download_pre_release(conf: Secrets) -> Generator[tuple[Path, Path], None, None]:
+    """
+    Downloads and unzips the pre-release archive. Returns the paths to the temporary
+    files of the project wheel and the SLC.
+
+    Usage:
+    with download_pre_release(conf) as unzipped_files:
+        project_wheel, slc_tar_gz = unzipped_files
+        ...
+    """
+
+    zip_url = conf.get(CKey.text_ai_pre_release_url)
+    if not zip_url:
+        raise ValueError("Pre-release URL is not set.")
+    zip_password = conf.get(CKey.text_ai_zip_password)
+    if not zip_password:
+        raise ValueError("Pre-release zip password is not set.")
+
+    # Download the file
+    response = requests.get(zip_url, stream=True)
+    response.raise_for_status()
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
```

Comment:
why `delete=False` ?

Reply:
Otherwise the temp file will be deleted on close(). I need to read from it.

Reply:
Actually, you are absolutely right. The file will never be deleted. I fixed it.